engine/src/lib.rs

```diff
@@ -142,6 +143,19 @@ mod contract {
         io.return_output(state.owner_id.as_bytes());
     }
 
+    /// Set owner account id for this contract.
+    #[no_mangle]
+    pub extern "C" fn set_owner() {
+        let mut io = Runtime;
+        let mut state = state::get_state(&io).sdk_unwrap();
+        require_owner_only(&state, &io.predecessor_account_id());
+        let args: SetOwnerArgs = io.read_input_borsh().sdk_unwrap();
```

Comment:
It is worth thinking about how we want the input and output to be encoding. In my opinion, the main factor to consider is what other tooling will be used with this `set_owner` function. For example, JSON encoding is standard in the Near ecosystem, so if this method will be called from Near CLI (even if that might be indirect by passing arguments to sputnik DAO) then it might be easier for users if the input and output were encoded with JSON instead of borsh.

Reply:
I also have an opinion that we should use JSON encoding here. 
cc @joshuajbouw

Reply:
I agree.

Reply:
Not yet. Because it isn't finished yet. But I have plans to add it this week for the SILO contract.

Reply:
I think I just need to change borsh input to json.

Reply:
Yes. You're right.
```rust
let args: SetOwnerArgs = serde_json::from_slice(&io.read_input().to_vec())
            .map_err(Into::<ParseTypeFromJsonError>::into)
            .sdk_unwrap();
```